spec/factories/goods_nomenclature_factory.rb

```diff
@@ -38,6 +38,10 @@
     after(:create) do |gono, _evaluator|
       if gono.associations[:goods_nomenclature_indents]
         gono.associations[:goods_nomenclature_indents].all?(&:save)
+        # We clear associations that rely on nested set that might have been
+        # loaded before the nested set tree node view was refreshed
+        gono.associations.clear
```

Comment:
Would it make sense to only clear the tree nodes and do it after the refresh has happened?

Reply:
We could but it makes no difference.